zebra-chain/src/orchard/tree.rs

```diff
@@ -221,6 +222,7 @@ pub struct NoteCommitmentTree {
     /// has non-empty nodes. Upper (near root) empty nodes of the branch are not
     /// stored.
     inner: bridgetree::Frontier<Node, { MERKLE_DEPTH as u8 }>,
+    cached_root: Cell<Option<Root>>,
```

Comment:
Missing Docs: Can you document the interior mutability on the `cached_root`, every method that uses it, and the type?

It might also be helpful to explain the impact of `cached_root`'s  unsynchronised interior mutability on the type.
(Unlike most types, immutable `NoteCommitmentTree` references can't be shared between threads. Neither can immutable references to anything containing a `NoteCommitmentTree` or its references.)

https://doc.rust-lang.org/nomicon/send-and-sync.html

Can you add similar docs to sapling as well?

Reply:
Added in 3478d76b